File: core/runtime/semantic_execution_scheduler.py
from __future__ import annotations

from typing import Any, Dict, List

from core.runtime.semantic_scheduler import schedule_semantic_tasks


def schedule_execution(tasks: List[Dict[str, Any]], max_tasks: int = 32) -> List[Dict[str, Any]]:
    return schedule_semantic_tasks(tasks, max_tasks=max_tasks)

File: src/runtime/semanticExecutionScheduler.ts
/**
 * Converted from Python: core/runtime/semantic_execution_scheduler.py
 * @generated — WebWeaveX python→javascript library port
 */

import { scheduleSemanticTasks } from "./semanticScheduler.js";

export function scheduleExecution(tasks: any, max_tasks: any = 32): any {
  return scheduleSemanticTasks(tasks, max_tasks);
}
export { scheduleSemanticTasks };
